app/resources/v1/article_resource.rb

```diff
@@ -30,8 +30,15 @@ def fetchable_fields
     super - [:cover_photo]
   end
 
-  def self.creatable_fields(_context)
-    %i[title content publicly_visible group cover_photo]
+  def self.creatable_fields(context)
+    attributes = %i[title content publicly_visible group cover_photo]
+
+    attributes += [:pinned] if update_permission?(context)
+    attributes
+  end
+
+  def self.update_permission?(context)
+    context[:user]&.permission?(:update, _model_class)
```

Comment:
Can't you use `update_permission?`? Not sure if that is available at the `self` context, but just wanted to check

Reply:
I tried to but it was indeed not available

Reply:
Is this method correct by the way? I wasn't sure what to use to see if someone has article.update permissions.

Reply:
Yes it is

Reply:
This is also done in the user resource btw (https://github.com/csvalpha/amber-api/blob/7c244de1a2ad1540794e99214f293ec0dbabc874/app/resources/v1/user_resource.rb#L115-L117)